amps/ags/rm-automation/rm-automation-community-rest-api/src/test/java/org/alfresco/rest/rm/community/hold/AddToHoldsV1Tests.java

```diff
@@ -0,0 +1,135 @@
+import org.alfresco.utility.model.SiteModel;
+import org.alfresco.utility.model.UserModel;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+/**
+ * V1 API tests for adding content/record folder/records to holds
+ *
+ * @author Damian Ujma
+ */
+public class AddToHoldsV1Tests extends BaseRMRestTest
+{
+    private static final String HOLD = "HOLD" + generateTestPrefix(AddToHoldsV1Tests.class);
+    private String holdNodeRef;
+    private FileModel documentHeld;
+    private final List<UserModel> users = new ArrayList<>();
+
+    @Autowired
+    private RoleService roleService;
+    @Autowired
+    private ContentActions contentActions;
+
+    @BeforeClass(alwaysRun = true)
+    public void preconditionForAddContentToHold()
+    {
+        STEP("Create a hold.");
+        Hold hold = createHold(FILE_PLAN_ALIAS, Hold.builder().name(HOLD).description("Description").reason("No reason").build(), getAdminUser());
+        holdNodeRef = hold.getId();
+        STEP("Create test files.");
+        SiteModel testSite = dataSite.usingAdmin().createPublicRandomSite();
+        documentHeld = dataContent.usingAdmin().usingSite(testSite)
+            .createContent(CMISUtil.DocumentType.TEXT_PLAIN);
+        dataContent.usingAdmin().usingSite(testSite)
+            .createContent(CMISUtil.DocumentType.TEXT_PLAIN);
+        dataContent.usingAdmin().usingSite(testSite)
+            .createContent(CMISUtil.DocumentType.TEXT_PLAIN);
+
+        STEP("Add the content to the hold.");
+        getRestAPIFactory()
+            .getHoldsAPI(getAdminUser())
+            .addChildToHold(HoldChild.builder().id(documentHeld.getNodeRefWithoutVersion()).build(), hold.getId());
+
+        STEP("Create users");
+        UserModel userAddHoldPermission = roleService.createUserWithSiteRoleRMRoleAndPermission(testSite,
+            UserRole.SiteCollaborator, holdNodeRef, UserRoles.ROLE_RM_MANAGER, PERMISSION_FILING);
+        users.add(userAddHoldPermission);
+    }
+
+    @Test
+    public void retrieveTheContentOfTheHoldUsingV1API() throws Exception
+    {
+        STEP("Retrieve the list of children from the hold and collect the entries that have the name of the active " +
+            "content held");
+        List<RestNodeModel> documentsHeld = restClient.authenticateUser(getAdminUser()).withCoreAPI()
+            .usingNode(toContentModel(holdNodeRef))
+            .listChildren().getEntries().stream()
+            .filter(child -> child.onModel().getName().contains(documentHeld
+                .getName()))
+            .collect(Collectors.toList());
+        STEP("Check the list of active content");
+        assertEquals(documentsHeld.size(), 1, "The active content is not retrieve when getting the children from the " +
+            "hold folder");
+        assertEquals(documentsHeld.get(0).onModel().getName(), documentHeld.getName());
```

Comment:
I think we should just get the set of names of held documents and assert:
```
assertEquals(documentNames, Set.of(documentHeld.getName()));
```

Reply:
Done.